src/core/models/answersconfig.js

```diff
@@ -0,0 +1,216 @@
+
+    /**
+     * Optional, prints full Answers error details when set to \`true\`. Defaults to false.
+     * @type {boolean}
+     */
+    this.debug = parsedConfig.debug;
+
+    /**
+     * Optional, If true, the search session is tracked. If false, there is no tracking. Defaults to true.
+     * @type {boolean}
+     */
+    this.sessionTrackingEnabled = parsedConfig.sessionTrackingEnabled;
+
+    /**
+     * Optional, invoked when the state of any component changes
+     * @type {function()}
+     */
+    this.onStateChange = parsedConfig.onStateChange;
+
+    /**
+     * Optional, analytics callback after a vertical search, see onVerticalSearch Configuration for additional details
+     * @type {function()}
+     */
+    this.onVerticalSearch = parsedConfig.onVerticalSearch;
+
+    /**
+     * Optional, analytics callback after a universal search, see onUniversalSearch Configuration for additional details
+     * @type {function()}
+     */
+    this.onUniversalSearch = parsedConfig.onUniversalSearch;
+
+    /**
+     * Optional, opt-out of automatic css variable resolution on init for legacy browsers
+     * @type {boolean}
+     */
+    this.disableCssVariablesPonyfill = parsedConfig.disableCssVariablesPonyfill;
+
+    /**
+     * Optional, the analytics key describing the Answers integration type. Accepts 'STANDARD' or 'OVERLAY', defaults to 'STANDARD'
+     * @type {string}
+     */
+    this.querySource = parsedConfig.querySource;
+
+    this.mock = parsedConfig.mock;
+    this.analyticsOptions = parsedConfig.analyticsOptions;
+    this.fieldFormatters = parsedConfig.fieldFormatters;
+
+    this.validate();
+    Object.freeze(this);
+  }
+
+  /**
+   * Validates the Answers config object to ensure things like api key and experience key are
+   * properly set.
+   */
+  validate () {
+    if (typeof this._apiKey !== 'string') {
+      throw new Error('Missing required `apiKey`. Type must be {string}');
+    }
+
+    if (typeof this.onStateChange !== 'function') {
+      throw new Error('onStateChange must be a function. Current type is: ' + typeof this.onStateChange);
+    }
+
+    if (typeof this.experienceKey !== 'string') {
+      throw new Error('Missing required `experienceKey`. Type must be {string}');
+    }
+
+    if (typeof this.onVerticalSearch !== 'function') {
+      throw new Error('onVerticalSearch must be a function. Current type is: ' + typeof this.onVerticalSearch);
+    }
+
+    if (typeof this.onUniversalSearch !== 'function') {
+      throw new Error('onUniversalSearch must be a function. Current type is: ' + typeof this.onUniversalSearch);
+    }
+  }
+
+  /**
+   * The Yext Answers API Key. If the apiKey contains the sandbox prefix, then it is removed leaving only the API key.
+   * @returns {string}
+   */
+  get apiKey () {
+    return this._apiKey.replace(sandboxPrefix, '');
+  }
+
+  /**
+   * The internal Yext Environment.
+   *  Defaults to Production
```

Comment:
nit: is this extra space here intentional?

Reply:
Probably not, I think it's a go convention that I do out of habit